### PR TITLE
Update dataset to use arXiv data

### PR DESCRIPTION
**Pull request by tylerhutcherson**

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -1,85 +1,145 @@
...
+from qna.db import get_cache, get_vectorstore
+from qna.prompt import basic_prompt
+from qna.data import get_olympics_docs, get_arxiv_docs
+from qna.constants import REDIS_URL
 
-
-@st.cache_resource
-def startup_qna_backend():
-    return make_qna_chain()
+def arxiv_index(_arxiv_documents):
```
Comment (tylerhutcherson) [inline app/app.py]: is this used?

File: app/app.py [hunk 2]
```diff
@@ -1,85 +1,145 @@
...
 
 from dotenv import load_dotenv
 load_dotenv()
 
 from urllib.error import URLError
-from qna.llm import make_qna_chain, get_cache
+from qna.llm import make_qna_chain
+from qna.db import get_cache, get_vectorstore
+from qna.prompt import basic_prompt
+from qna.data import get_olympics_docs, get_arxiv_docs
```
Comment (tylerhutcherson) [inline app/app.py]: do we need the olympics data anymore?

File: app/app.py [hunk 3]
```diff
@@ -1,85 +1,145 @@
...
+        st.slider("Number of Tokens", 100, 500, 400, key="max_tokens")
+        st.slider("Distance Threshold", .1, .9, .5, key="distance_threshold", step=.1)
+        st.button("Clear Chat", key="clear_chat", on_click=lambda: st.session_state['messages'].clear())
+        st.button("Clear Cache", key="reset", on_click=reset_app)
+
+    st.write("## Arxiv Document Chat Application")
+
+    st.write("**Put in a topic area and a question within that area to get an answer!**")
+    topic = st.text_input("Topic Area", key="arxiv_topic")
+    papers = st.number_input("Number of Papers", key="num_papers", value=10, min_value=1, max_value=50, step=2)
```
Comment (tylerhutcherson) [inline app/app.py]: This seems like a hard parameter for a user to know "apriori"

File: app/app.py [hunk 4]
```diff
@@ -1,85 +1,145 @@
...
+
+
+    if st.button("Chat!"):
+        if is_updated(topic):
+            st.session_state['previous_topic'] = topic
+            with st.spinner("Loading information from Arxiv to answer your question..."):
+                create_arxiv_index(st.session_state['arxiv_topic'], st.session_state['num_papers'], prompt)
```
Comment (tylerhutcherson) [inline app/app.py:116]: How long does this process take? Is there a risk of resource exhaustion w.r.t embedding creation?

File: app/qna/data.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+from typing import List
+
+from langchain.schema import Document
+from langchain.document_loaders import ArxivLoader
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+def get_olympics_docs() -> List[Document]:
```
Comment (tylerhutcherson) [inline app/qna/data.py]: can we remove?

File: app/qna/db.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+from langchain.schema import Document
+from langchain.vectorstores.redis import Redis as RedisVDB
+
+from qna.llm import get_embeddings
+from qna.constants import CACHE_TYPE, INDEX_NAME, REDIS_URL
+
+def get_cache():
+    # construct cache implementation based on env var
+    if CACHE_TYPE == "semantic":
```
Comment (tylerhutcherson) [inline app/qna/db.py:12]: we could simplify this too and only support semantic cache?

File: app/qna/llm.py [hunk 7]
```diff
@@ -33,114 +33,63 @@ def get_embeddings() -> Embeddings:
         #currently Azure OpenAI embeddings require request for service limit increase to be useful
         #using build-in HuggingFace instead
         from langchain.embeddings import HuggingFaceEmbeddings
-        embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
+        embeddings = HuggingFaceEmbeddings(model_name=HUGGINGFACE_MODEL_NAME)
```
Comment (tylerhutcherson) [inline app/qna/llm.py]: still wondering here if it makes sense to even try to do the whole "choose your LLM" thing. We could keep it simple by requireing only OpenAI direct.. folks could adapt and fork as they want?

**Review by tylerhutcherson — COMMENTED** [review]
A couple questions but I think this is the right direction for this app. The other thing we'd need is a major update to the readme. (screen shots, explanations)

**Comment by Spartee** [comment]
Merged in #12

**Pull request closed without merging**